en/middleware/micrortps_throughput_test.md

```diff
@@ -0,0 +1,105 @@
+
+> **Tip** This example requires that you [Manually Generate Client and Agent Code](../middleware/micrortps_manual_code_generation.md).
+
+## Create the uORB message
+
+First create a new uORB message for this test in the folder **/Firmware/msg/**. The message file will be called **throughput_256.msg** and have the following content:
+
+```text
+uint8[256] data
+```
+
+This can be done with the command line below:
+
+```sh
+cd /path/to/PX4/Firmware/msg
+echo uint8[256] data > throughput_256.msg
+```
+
+Register the new message adding it to the list of messages in the file: **/Firmware/msg/CMakeLists.txt**:
+
+```cmake
+...
+wind_estimate.msg
+throughput_256.msg
+)
+...
+```
+
+Give the message a topic id by adding a line in the **/Firmware/Tools/message_id.py** script:
+
+```python
+...
+    'wind_estimate': 94,
+    'throughput_256': 95,
+}
+...
+```
+
+## Disable automatic bridge code generation
+
+Disable automatic generation of bridge code (as part of the PX4 build process) by setting the variable `GENERATE_RTPS_BRIDGE` to `off` in the *.cmake* file for the target platform (*cmake/configs/*):
+
+```sh
+set(GENERATE_RTPS_BRIDGE off)
+```
+
+
+## Generate the bridge code
+
+Manually generate bridge code using *generate_microRTPS_bridge.py* (the code will send and receive "just" our `throughput_256` uORB topic):
+
+```sh
+cd /path/to/PX4/Firmware
+python Tools/generate_microRTPS_bridge.py --send msg/throughput_256.msg --receive msg/throughput_256.msg
+```
+
+The *Client* source code is generated in **src/modules/micrortps_bridge/micrortps_client/** and the *Agent* in **src/modules/micrortps_bridge/micrortps_agent/**.
+
+### Modify the client code
+
+Next we modify the *Client* to send a *throughput_256* message on every loop. This is required because the topic is not actually being published by PX4, and because we want to ensure that it is sent at the greatest possible rate.
+
+Open the file **src/modules/micrortps_bridge/micrortps_client/microRTPS_client.cpp**. Update the `while` loop in the `send()` function to look like this:
+
+```cpp
+...
+while (!_should_exit_task)
+{
+    //bool updated;
+    //orb_check(fds[0], &updated);
+    //if (updated)
+    {
+        // obtained data for the file descriptor
+        struct throughput_256_s data = {};
+        // copy raw data into local buffer
+        //orb_copy(ORB_ID(throughput_256), fds[0], &data);
+        data.data[0] = loop%256;
+        serialize_throughput_256(&data, data_buffer, &length, &microCDRWriter);
+        if (0 < (read = transport_node->write((char)95, data_buffer, length)))
+        {
+            total_sent += read;
+            ++sent;
+        }
+    }
+     usleep(_options.sleep_ms*1000);
+    ++loop;
+}
+...
+```
+
+Compile and launch both the [client](../middleware/micrortps.md#client-px4-firmware) and the [agent](../middleware/micrortps.md#agent-off-board-fastrtps-interface).
```

Comment:
@hamishwillee When we generate the code manually we allow to compile and upload this code to the device in a 'normal' way but disabling the auto generation as is talled in the throughput test example.